public/app/percona/dbaas/components/DBCluster/DBClusterStatus/DBClusterStatus.tsx

```diff
@@ -15,8 +17,18 @@ import { DBClusterStatusProps } from './DBClusterStatus.types';
 import { getProgressMessage, getShowProgressBarValue } from './DBClusterStatus.utils';
 
 export const DBClusterStatus: FC<DBClusterStatusProps> = ({ dbCluster, setSelectedCluster, setLogsModalVisible }) => {
-  const { message, finishedSteps, totalSteps } = dbCluster;
-  const status = dbCluster.status as Status;
+  const { result: clusters = {}, loading } = useSelector(getPerconaDBClustersDetails);
+  const { result: dbClusters = [] } = useSelector(getPerconaDBClusters);
+
+  const {
+    status,
+    totalSteps = 0,
+    finishedSteps = 0,
+    message,
+  } = Object.keys(clusters).length && dbCluster.id && dbClusters.length === Object.keys(clusters).length
```

Comment:
I'm seeing the same condition twice, do you think it would make sense to have an util function to handle it?

Reply:
hmm... so we'll get there type error, we can solve it... but i think simplier just left 2 conditions

![image](https://user-images.githubusercontent.com/90199600/196700425-1a41ed0b-74c3-43cd-bb42-c6f1aa44b78e.png)

Reply:
Let's move this to the refactoring stage.